### PR TITLE
Nlp evaluation V2

### PR DESCRIPTION
**Pull request by aaarrti**
### Description
- Adding metrics implementations for text-classification.

### Implemented changes
 -  [x] Sensitivity and Relative Stability metrics implemented.
 -  [x]  MPR and random logit implemented.
 -  [x]  Implemented Token Flipping (analogue of pixel flipping), e.g., used here https://arxiv.org/pdf/2202.07304.pdf in section 6.2., paragraph 3.
 -  [x]  Implementations + wrappers for XAI methods.
    - I will look later on into the possibility of adding LRP-based explanations. 
 -  [x] API export under `quantus.nlp`, with new evaluate function, with smaller number of arguments and type aliases commonly used.
 - [x] API documentation
 - [x] Notebook(s) showcasing new sub-package
 - [x] Basic tests covering calls with expected parameters
 - [x] Drop python 3.7 support, add python 3.10 support
 - [x] Upgrade dependencies

### Knows issues
- Test now take far too long to run 
- Some of tests fails on python 3.10, some fail time to time
- Model Parameter Randomisation crashes GitHub runner with OOM error
- PR is already huge, and it is far from including everything planned 
- I'm not 100% certain in quality of my implementation of torch gradient-based XAI methods.
- `Softmax`, `return_nan_when_predictions_change`, support for non HF subclassed model not implemented (since PR is huge, I'd rather do them in separate one on top)

### CODE REVIEW

**Comment by aaarrti** [comment]
@annahedstroem Hi, this PR is still a work in progress, but I would appreciate it if you could start reviewing parts of it, e.g., `quantus.nlp.helpers.model` or base metric class `quantus.nlp.metrics.BatchedMetric`. Or anything else you'd like 🙃 .
There are still a few issues with tests, I will be resolving them over the following days (including coverage). I will also add doc strings, notebooks, etc. soon. I did not implement `softmax`, `return_aggregate` and `return_nan_when_prediction_changes` behaviours. Since the PR is huge, I'd rather add them later on in a separate one.

**Comment by aaarrti** [comment]
I will be trying to split the PR in few smaller ones.

**Comment by aaarrti** [comment]
branch was messed up 😢
